package.json

```diff
@@ -21,20 +20,47 @@
     "javascript",
     "styleguide"
   ],
-  "peerDependencies": {
-    "babel-eslint": "^10.0.1",
-    "eslint": "^5.12.0",
-    "eslint-config-airbnb": "^17.1.0",
-    "eslint-plugin-eslint-comments": "^3.1.2",
-    "eslint-plugin-import": "^2.14.0",
-    "eslint-plugin-jest": "^22.17.0",
-    "eslint-plugin-jsx-a11y": "^6.1.2",
-    "eslint-plugin-react": "^7.12.3",
-    "eslint-plugin-backpack": "^0.2.2",
-    "eslint-plugin-skyscanner-dates": "^1.0.9"
-  },
+  "bin": "main.js",
+  "main": "index.js",
   "scripts": {
+    "format:md": "prettier --write **/*.md",
+    "format:packagejson": "format-package -w",
+    "lint": "npm run lint:js",
+    "lint:js": "eslint . --ext js",
+    "lint:js:fix": "eslint . --ext js --fix",
+    "postinstall": "node ./main.js",
     "pretest": "npx ensure-node-env",
     "test": "(cd test && ./e2e-test.sh)"
+  },
+  "husky": {
+    "hooks": {
+      "pre-commit": "lint-staged"
+    }
+  },
+  "lint-staged": {
+    "**/!(package|package-lock).json": "prettier --write",
+    "*.js": "npm run lint:js:fix",
+    "*.md": "prettier --write",
+    "package.json": "npm run format:packagejson"
+  },
+  "dependencies": {
```

Comment:
I'm using the dependencies defined in ``, but many of them are old:
```
Package                 Current   Wanted  Latest  Location
eslint                   5.16.0   5.16.0   6.8.0  eslint-config-skyscanner
eslint-config-airbnb     17.1.1   17.1.1  18.0.1  eslint-config-skyscanner
eslint-config-prettier    3.6.0    3.6.0  6.10.0  eslint-config-skyscanner
eslint-plugin-jest      22.21.0  22.21.0  23.6.0  eslint-config-skyscanner
```

I think we should upgrade `eslint`, `eslint-config-prettier` and `eslint-plugin-jest` to the latest version (I see there's a pr upgrading `eslint-config-airbnb`). What do you think about this?

Reply:
@olliecurtis what do you think about upgrading `eslint`, `eslint-config-prettier` and `eslint-plugin-jest` to the latest version? Given this will already be a major release, I would update them :-)

Reply:
I think this makes sense and we should update to align with the latest versions, we just need to be aware of the other packages if they have peer/dependencies on older versions of these it e.g. `eslint-config-backpack` it may give console warnings that there is incompatible versions, but if it all checks out I am happy to upgrade them as we should probably keep up to date and adjust the rules if there are changes. 

Which testing in projects should catch any burning problems

Reply:
I've updated `eslint-config-prettier` for now.

When updating `eslint`, `eslint-config-airbnb` complains (this can be done as part of https://github.com/Skyscanner/eslint-config-skyscanner/pull/31)

Updating `eslint-plugin-jest` I get `npm WARN tsutils@3.17.1 requires a peer of typescript@>=2.8.0 || >= 3.2.0-dev || >= 3.3.0-dev || >= 3.4.0-dev || >= 3.5.0-dev || >= 3.6.0-dev || >= 3.6.0-beta || >= 3.7.0-dev || >= 3.7.0-beta but none is installed. You must install peer dependencies yourself.`. I'll try looking into it later 😅